src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -272,6 +282,20 @@
             type="warehouse_size",
         ),
     ),
+    RefreshModeType=OneOf(
+        MultiStringParser(
+            snowflake_dialect.sets("refreshmode_types"),
+            KeywordSegment,
+            type="refreshmode_type",
+        )
+    ),
```

Comment:
Why not just this?:

```suggestion
    RefreshModeType=OneOf(
        "AUTO",
        "FULL",
        "INCREMENTAL"
    ),
```

The main difference is you cannot set `type="refreshmode_type"` in this way, but unless you explicitly need to refer to this elsewhere (e.g. you have a rule to pick out this particular syntax) that's usually not needed.

I say this has been done for a lot of other Snowflake syntax (`compression_types`, `files_types`...etc.) but not entirely sure it's necessary, or that it's something that should be continued....

Reply:
I have also tried your second suggestion, and it also fails, particularly on the `Match Pattern Received` stage `generate-fixture-yml`.  It looks like without the MultiStringParser argument, it's interpreting in some other way (maybe this one can be described in more detail somewhere in the doc).

Onto your second suggestion for removing the `type` part, I have removed it and no new issues have arisen, the only reference is in the TC `create_table.yml` content, it had switched the `AUTO` object key `refreshmode_type` to only `keyword`. 

In regard to the question if it should be continued, it really depends if it's going to affect the normal behaviour of the parser from the snowflake side, I'm not 100% sure even if the TCs are covering this aspect at all, but still we can put it on the backlog as part of refactoring the code :)